h2o-py/tests/testdir_algos/automl/target_encoding/pyunit_target_encoding_model.py

```diff
@@ -62,6 +63,19 @@ def test_target_encoding_fit_method():
     assert len(transformed.col_names) == len(expected_columns)
     assert sorted(transformed.col_names) == sorted(expected_columns) # 4 encoded columns
 
+    gbm_with_te=H2OGradientBoostingEstimator(score_tree_interval=10,
+                                             ntrees=500,
+                                             sample_rate=0.8,
+                                             col_sample_rate=0.8,
+                                             seed=1234,
+                                             stopping_rounds=5,
+                                             stopping_metric="AUC",
+                                             stopping_tolerance=0.001,
+                                             model_id="gbm_with_te")
+
+    myX = ["pclass", "sex", "age", "sibsp", "parch", "fare", "cabin_te", "embarked_te", "home.dest_te"]
+    gbm_with_te.train(x=myX, y=targetColumnName, training_frame=transformed)
```

Comment:
@Pscheidl could you please take a look at this test? I added gbm estimator and trying to train it with new encodings. Training of the model fails when we are using frame which was encoded. Looks like frame is still being locked. May I be missing something?

Reply:
Yes, I'll try to help. Release duties today, so I'll have to postpone this to today evening CET or weekend.

Reply:
created a jira https://0xdata.atlassian.net/browse/PUBDEV-7434

Reply:
@Pscheidl did you have a chance to verify whether there is a bug?

Reply:
Not yet, but I can promise today.

Reply:
I pushed the fix. You can always revert if you don't like it, it modifies your code.

Reply:
@Pscheidl This PR is about demo and I think fix should be provided separately in the jira that I have created as it is a bug in a new version API. Same code is working with old API and there are tests that prove it (MichalK suggested me this piece of code and probably it is not incorrect, it just did not take into account your changes). Fix quite often could be placed into different places.

Reply:
I'll fix it in different PR then.

Reply:
Thank you, @Pscheidl